src/lib.rs

```diff
@@ -2722,3 +2706,79 @@ fn command_add_output_file(
         cmd.arg("-o").arg(&dst);
     }
 }
+
+static NEW_STANDALONE_ANDROID_COMPILERS: [&str; 4] = [
+    "aarch64-linux-android21-clang",
+    "armv7a-linux-androideabi16-clang",
+    "i686-linux-android16-clang",
+    "x86_64-linux-android21-clang",
```

Comment:
Are these embedded version numbers something that will change with each release? Is there perhaps other env vars we can probe for or a well-known executable which doesn't change which we can look for and then find sibling compilers?

Reply:
> Are these embedded version numbers something that will change with each release?

No, this is minimum API level. 16 for armv7a, i686 and 21 for x86_64 /aarch64 is minium API version. Android 4.0.3 (API level 15) is not supported by this new clang toolchain. But I don't think that this is problem, rustc will also drop support for such low API level rust-lang/rust#71123 .

So this numbers changed when Google drop support of 4.0.3 in their SDK/NDK, have no idea when this happens.

Reply:
Hm ok, is there perhaps an official location where we can point to which has the names of these various compilers?

Reply:
https://android.googlesource.com/platform/ndk/+/refs/heads/ndk-release-r21/docs/BuildSystemMaintainers.md#Clang

See 
> Second, a target-specific Clang can be used. In addition to the clang and clang++ binaries, there are also <triple><API-level>-clang and <triple><API-level>-clang++ scripts. For example, to target API 26 32-bit ARM, invoke armv7a-linux-androideabi26-clang or armv7a-linux-androideabi26-clang++ instead of clang or clang++.

Reply:
I added this link into code's comments.